[Security Solution][Data Quality Dashboard][Serverless] Fix fetchAvailableIndices in get_index_stats

kapral18 opened the pull request:
addresses #196528

- Remove unused get_available_indices.ts params helper file.
- Change fetchAvailableIndices to use creation_date from _cat api instead of targeting @timestamp field of indices

## UI Changes:

Before:
![image](https://github.com/user-attachments/assets/1954a8b6-1760-4ec7-b3d3-167b724f8b3c)
After:
![image](https://github.com/user-attachments/assets/232674a1-9691-4d49-862e-99873f22c3cf)


<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

kapral18 committed:
[Security Solution][Data Quality Dashboard][Serverless] Fix fetchAvailableIndices in get_index_stats

addresses #196528

- Remove unused get_available_indices.ts params helper file.
- Change fetchAvailableIndices to use creation_date from _cat api
  instead of targeting @timestamp field of indices

elasticmachine commented:
Pinging @elastic/security-threat-hunting-explore (Team:Threat Hunting:Explore)

angorayc commented on x-pack/plugins/ecs_data_quality_dashboard/server/lib/fetch_available_indices.ts:
```diff
@@ -4,18 +4,70 @@
+    return creationDateMillis >= startDateMillis && creationDateMillis <= endDateMillis;
+  });
+
+  return {
+    aggregations: {
+      index: {
+        buckets: filteredIndices.map((indexInfo) => ({
+          key: indexInfo.index,
+        })),
+      },
+    },
+  };
```
I think we can make it simpler and return an array here, as we are no longer doing any aggregation in the query
```
filteredIndices.map((indexInfo) => ({
          key: indexInfo.index,
        }))
```

angorayc replied:
Can we also deploy this PR to the cloud and check it again before merge?

kapral18 replied:
Yes also saw that after push and started simplifying. Thanks for catching tho.
I will add the label. Thanks

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

kapral18 committed:
refactor: simplify, return string instead of aggregation

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/244326)
* Commit: c9c621889e96977c46c79d116f80bc3a49f7bda6
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-197065-c9c621889e96`

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/244326","number":244326,"commit":"c9c621889e96977c46c79d116f80bc3a49f7bda6"}],"number":244326}
buildkite-pr-comment-->

kapral18 committed:
fix(fetch_available_indices): include indices with data within date range regardless of creation date

Previously, `fetchAvailableIndices` only considered indices whose creation dates fell within the specified date range. This missed indices that were created outside the date range but contained data within it. This fix updates the function to also include indices that have data within the specified date range by performing an additional search aggregation.

- Return back `getRequestBody` helper to build the search request for fetching indices with data in range.
- Improve error handling for invalid date formats with more specific messages.
- Update tests to reflect the new logic and error messages.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/246131)
* Commit: 67b898996db929ac25ba783ee80bef73fc88dca5
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-197065-67b898996db9`

### History
* :green_heart: [Build #244326](https://buildkite.com/elastic/kibana-pull-request/builds/244326) succeeded c9c621889e96977c46c79d116f80bc3a49f7bda6

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246131","number":246131,"commit":"67b898996db929ac25ba783ee80bef73fc88dca5"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/244326","number":244326,"commit":"c9c621889e96977c46c79d116f80bc3a49f7bda6"}],"number":246131}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/246227)
* Commit: a29630f4956a1231493f65f3ff4fe7d0543a6b98
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-197065-a29630f4956a`

### History
* :green_heart: [Build #246131](https://buildkite.com/elastic/kibana-pull-request/builds/246131) succeeded 67b898996db929ac25ba783ee80bef73fc88dca5
* :green_heart: [Build #244326](https://buildkite.com/elastic/kibana-pull-request/builds/244326) succeeded c9c621889e96977c46c79d116f80bc3a49f7bda6

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246227","number":246227,"commit":"a29630f4956a1231493f65f3ff4fe7d0543a6b98"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246131","number":246131,"commit":"67b898996db929ac25ba783ee80bef73fc88dca5"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/244326","number":244326,"commit":"c9c621889e96977c46c79d116f80bc3a49f7bda6"}],"number":246227}
buildkite-pr-comment-->

semd reviewed (approved):
LGTM

angorayc reviewed (approved):
Thanks for fixing this 👍 👍 👍

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/247283)
* Commit: 64ea9110da29345159d1e64bda869c97ac461352
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-197065-64ea9110da29`

### Failed CI Steps
* [FTR Configs #28](https://buildkite.com/elastic/kibana-pull-request/builds/247283#0192df7a-abaf-4aa4-89b9-ff76c06364bb)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/247283#0192df7a-abaf-4aa4-89b9-ff76c06364bb) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/247283/jobs/0192df7a-abaf-4aa4-89b9-ff76c06364bb/artifacts/0192df9c-5dc2-448d-b69a-1ab616e35f6d) FTR Configs #<span></span>28 / saved objects tagging - functional tests visualize integration editing allows to assign tags to an existing visualization

### History
* :green_heart: [Build #246227](https://buildkite.com/elastic/kibana-pull-request/builds/246227) succeeded a29630f4956a1231493f65f3ff4fe7d0543a6b98
* :green_heart: [Build #246131](https://buildkite.com/elastic/kibana-pull-request/builds/246131) succeeded 67b898996db929ac25ba783ee80bef73fc88dca5
* :green_heart: [Build #244326](https://buildkite.com/elastic/kibana-pull-request/builds/244326) succeeded c9c621889e96977c46c79d116f80bc3a49f7bda6

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/247283","number":247283,"commit":"64ea9110da29345159d1e64bda869c97ac461352"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246227","number":246227,"commit":"a29630f4956a1231493f65f3ff4fe7d0543a6b98"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246131","number":246131,"commit":"67b898996db929ac25ba783ee80bef73fc88dca5"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/244326","number":244326,"commit":"c9c621889e96977c46c79d116f80bc3a49f7bda6"}],"number":247283}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/11612495058

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/198525">](https://github.com/elastic/kibana/pull/198525)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->

kapral18 commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.16|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/200551">](https://github.com/elastic/kibana/pull/200551)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.8 -->
